src/Player.cpp

```diff
@@ -993,6 +993,58 @@ void Player::Update( float fDeltaTime )
 		}
 	}
 
+	// track held misses
+	{
+		float largestWindow = .0;
+		largestWindow = max(largestWindow, GetWindowSeconds(TW_W1));
+		largestWindow = max(largestWindow, GetWindowSeconds(TW_W2));
+		largestWindow = max(largestWindow, GetWindowSeconds(TW_W3));
+		largestWindow = max(largestWindow, GetWindowSeconds(TW_W4));
+		largestWindow = max(largestWindow, GetWindowSeconds(TW_W5));
+
+		const float rate = GAMESTATE->m_SongOptions.GetCurrent().m_fMusicRate;
+		const SongPosition songPosition = m_pPlayerState->m_Position;
+		const float musicPosition = songPosition.m_fMusicSeconds + (songPosition.m_LastBeatUpdate.Ago() * rate);
+		const int lastCheckRow = BeatToNoteRow(m_Timing->GetBeatFromElapsedTime(musicPosition + (largestWindow * rate)) + 1);
+
+		vector<bool> seenTracks(m_NoteData.GetNumTracks(), false);
+
+		NoteData::all_tracks_iterator iter = *m_pIterNeedsTapJudging;	// copy
```

Comment:
Any reason this part isn't in the for loop itself? iter doesn't seem to be used outside of it.

Reply:
I didn't add it in the for loop init because it makes the line super long and so that the comment can be there. I figured that's fine because it's all in local scope that ends a couple of lines further down anyway.

If you prefer it differently I can change it.

Reply:
I think you can break up the for loop and put it across multiple lines. I think that's perfectly fine. Something like:

```cpp
for(NoteData::all_tracks_iterator iter = *m_pIterNeedsTapJudging; // copy;
    !iter.IsAtEnd() && iter.Row() <= lastCheckRow; ++iter)
{

}
```
Fwiw I don't think the `// copy` comment is necessary. I think that's already implied.

Reply:
you can also use `auto` if you so like:

```cpp
for (auto iter = *m_pIterNeedsTapJudging; !iter.IsAtEnd() && iter.Row() <= lastCheckRow; ++iter)
{
}
```

Reply:
I completely forgot about auto. I used that!